### PR TITLE
Migrate storage file from .txt to .json

### PR DESCRIPTION
**Pull request by st3no**

### CODE REVIEW

**Comment by elrhomariyounes** [comment]
Thank you very  much @st3no. I love the new approach that you implemented.

Can you target this to dev branch.

File: YoCli/YoCli.csproj [hunk 1]
```diff
@@ -20,6 +20,7 @@
     <PackageReference Include="McMaster.Extensions.Hosting.CommandLine" Version="3.0.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.7" />
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="3.1.7" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
```
Comment (elrhomariyounes) [inline YoCli/YoCli.csproj]: Can we use Sytem.Text.Json instead of Newtonsoft.Json(2.5MB) keep it lighweight

File: YoCli/Commands/Export.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        public Task<int> OnExecute()
+        {
+            if (!Directory.Exists(Path))
+            {
+                ConsoleMessage.PrintError(_console, "Invalid path!");
+                return Task.FromResult(-1);
+            }
+
+            //File.Copy(storageFilePath, Path.Combine(filePathToExport, "notes.json"));
```
Comment (elrhomariyounes) [inline YoCli/Commands/Export.cs]: In case if the Directory Exists we copy the file with Path Argument

**Review by elrhomariyounes — CHANGES_REQUESTED** [review]

File: YoCli/Commands/Export.cs (hunk 2 above)
Comment (st3no) [reply YoCli/Commands/Export.cs]: I forgot to implement this command after refactoring :)

File: YoCli/YoCli.csproj (hunk 1 above)
Comment (st3no) [reply YoCli/YoCli.csproj]: That's good idea !

File: YoCli/Program.cs [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
-        {
-            _console = console;
-        }
-
-        public static async Task<int> Main(string[] args)
+        static async Task<int> Main(string[] args)
         {
             var builder = new HostBuilder().ConfigureServices((hostContext, services) =>
             {
-                services.AddSingleton<INoteService, NoteService>();
```
Comment (elrhomariyounes) [inline YoCli/Program.cs:25]: Hi @st3no, can we keep the implementation of NoteService and refactor the business logic in the commands (Read, Find, Write, Export, Import) to a another service that implement INoteService ? So you can add some unit tests for the implementation. Thank you very much

**Review by elrhomariyounes — CHANGES_REQUESTED** [review]

**Pull request closed without merging**